csharp/.vscode/launch.json

```diff
@@ -0,0 +1,65 @@
+{
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "name": "C#: Standalone Debug",
+            "type": "coreclr",
+            "request": "launch",
+            "preLaunchTask": "dotnet: build",
+            "program": "${workspaceFolder}/extractor/Semmle.Extraction.CSharp.Standalone/bin/Debug/net8.0/Semmle.Extraction.CSharp.Standalone.dll",
+            "args": [],
+            // Set the path to the folder that should be extracted:
+            "cwd": "${workspaceFolder}/ql/test/library-tests/standalone/standalonemode",
+            "env": {
+                "CODEQL_THREADS": "1",
+                "CODEQL_EXTRACTOR_CSHARP_OPTION_LOGGING_VERBOSITY": "progress+++",
+                "CODEQL_EXTRACTOR_CSHARP_OPTION_TRAP_COMPRESSION": "NONE",
+            },
+            "stopAtEntry": true,
+            "console": "internalConsole",
+            "justMyCode": false,
+            "symbolOptions": {
+                "searchPaths": [],
+                "searchMicrosoftSymbolServer": true,
+                "searchNuGetOrgSymbolServer": true
+            },
+            "sourceLinkOptions": {
+                "*": {
+                    "enabled": true
+                }
+            },
+            "suppressJITOptimizations": true
+        },
+        {
+            "name": "C#: Autobuild Debug",
+            "type": "coreclr",
+            "request": "launch",
+            "preLaunchTask": "dotnet: build",
+            "program": "${workspaceFolder}/autobuilder/Semmle.Autobuild.CSharp/bin/Debug/net8.0/Semmle.Autobuild.CSharp.dll",
+            // Set the path to the folder that should be extracted:
+            "cwd": "${workspaceFolder}/ql/integration-tests/all-platforms/autobuild",
+            "stopAtEntry": true,
+            "args": [],
+            "env": {
+                // The below folders need to exist before debugging
+                "CODEQL_EXTRACTOR_CSHARP_TRAP_DIR": "${workspaceFolder}/ql/integration-tests/DB",
+                "CODEQL_EXTRACTOR_CSHARP_SOURCE_ARCHIVE_DIR": "${workspaceFolder}/ql/integration-tests/DB",
+                "CODEQL_EXTRACTOR_CSHARP_DIAGNOSTIC_DIR": "${workspaceFolder}/ql/integration-tests/DB",
+                "CODEQL_EXTRACTOR_CSHARP_SCRATCH_DIR": "${workspaceFolder}/ql/integration-tests/DB",
+            }
+        },
+        {
+            "name": "C#: Binary Log Debug",
+            "type": "coreclr",
+            "request": "launch",
+            "preLaunchTask": "dotnet: build",
+            "program": "${workspaceFolder}/extractor/Semmle.Extraction.CSharp.Driver/bin/Debug/net8.0/Semmle.Extraction.CSharp.Driver.dll",
+            "stopAtEntry": true,
+            "args": [
+                "--binlog",
+                "${workspaceFolder}/ql/integration-tests/all-platforms/binlog/test.binlog"
+            ],
+            "env": {}
+        },
+    ]
+}
```

Comment:
Should we also have an entry for "attach"?
```
{
            "name": ".NET Core Attach",
            "type": "coreclr",
            "request": "attach",
            "processId": "${command:pickProcess}",
            "requireExactSource": false
        },
```

Reply:
I don't mind adding it, but I haven't used that lately. Also, attaching to the running extractor process likely requires some changes to your local sembuild commands too, to build a bundle with debug artifacts.

Reply:
Yes. It also looks like this code has changed a lot recently, so I am not sure how to change the build to produce debug artifacts. Maybe we just need to change Release -> Debug in `ql/csharp/scripts/create-extractor-pack.sh`.
So this is out of scope for this PR.